src/riak_ql_ddl.erl

```diff
@@ -179,22 +180,44 @@ get_local_key(?DDL{table = T}=DDL, Obj)
     Mod = make_module_name(T),
     get_local_key(DDL, Obj, Mod).
 
+-spec desc_adjusted_key([any()], module(), ?DDL{}) ->
```

Comment:
What would help alleviate my confusion, I suspect, and thus future programmers' confusion as well, would be at the very least to rethink this function name.

I also suspect this interface would be easier to understand if client code called `lk_to_pk` first without any negation happening, then passed it to this function, whatever it ends up being called.